src/mrack/providers/podman.py

```diff
@@ -58,8 +58,33 @@ async def init(self, container_images, default_network, ssh_key, container_optio
         logger.info(f"{self.dsp_name}: Init duration {login_duration}")
 
     async def prepare_provisioning(self, reqs):
-        """Prepare provisioning."""
-        pass
+        """Pull missing images to prepare provisioning for podman."""
+        image_list = set()
+        image_list.add("quay.io/neoave/snappeas:fedora-33")
+        for req in reqs:
+            image = req["image"]
+            if not await self.podman.image_exists(image):
+                logger.debug(f"{self.dsp_name}: Pull of image '{image}' required")
+                image_list.add(image)
+            else:
+                logger.debug(f"{self.dsp_name}: Image '{image}' present")
+
+        if not image_list:
```

Comment:
The success message that we have all the images could be moved after the `success = all(pull_results)`.  So that there would be some pointer that this part of operation ended.

Reply:
Can do